codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -337,7 +330,7 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> ou
         }
         case READ_CHUNKED_CONTENT: {
             assert chunkSize <= Integer.MAX_VALUE;
-            int toRead = Math.min((int) chunkSize, maxChunkSize);
+            int toRead =(int) chunkSize;
```

Comment:
No, it's a `long`. The name is kind of misleading, it might be better as `remainingContent` or something, though that then becomes misleading for `transfer-encoding: chunked` content...